# Version(V2.0)

## Overview

## Input
- PacBio HiFi data
- OmniC sequencing data

## Output

The final output corresponds to a diploid assembly, with the primary/alternate approach. 
We are generating 2 pseudo-haplotypes (primary and alternate). The primary assembly is more complete and consists of longer phased blocks. The alternate consists of haplotigs 
(contigs of clones with the same haplotype) in heterozygous regions and is not as complete and more fragmented.
As explained by Heng Li ([here](https://lh3.github.io/2021/04/17/concepts-in-phased-assemblies)), given the characteristics of the latter, it cannot be considered on its own, but as a complement of the primary 
assembly.

There are 3 main differences with respect to the previous version of the pipeline. 
- Input chromatin conformation capture data we use is Omni-C data, not Hi-C.
- The version of the *de novo* assembler, changed from 0.13 to 0.15.
- The workflow to generate the mitogenome assembly changed to MitoHiFi

## Software

Almost all of the links will send you to the corresponding repository. 
The rest of the links correspond to the main documentation web site of the tool.

### Assembly

| Purpose | Program | Version | 
|:-------|:---------|--------:|
| Filtering PacBio HiFi adapters |  [HiFiAdapterFilt](https://github.com/sheinasim/HiFiAdapterFilt) | Commit `64d1c7b` | 
| K-mer counting |  [meryl](https://github.com/marbl/meryl) | `1` | 
| Estimation of genome size and heterozygosity |  [GenomeScope](https://github.com/tbenavi1/genomescope2.0) | `2` |
| *de novo* assembly (contiging) |  [HiFiasm](https://github.com/chhylp123/hifiasm) |   `0.15-r327` |
| Long-read, genome-genome alignment |  [minimap2](https://github.com/lh3/minimap2) |  `2.16` | 
| Remove low-coverage, duplicated contigs | [purge_dups](https://github.com/dfguan/purge_dups) | `1.0.1` |
| HiC mapping for SALSA |  [Arima Genomics mapping pipeline](https://github.com/ArimaGenomics/mapping_pipeline) |  Commit `2e74ea4` | 
| HiC Scaffolding | [SALSA](https://github.com/marbl/SALSA) | `2` | 
| Gap closing |  [YAGCloser](https://github.com/merlyescalona/yagcloser) | Commit `20e2769` | 

### Omni-C Contact map generation

| Purpose | Program | Version | 
|:-------|:---------|--------:|
| Short-read alignment  | [bwa](https://github.com/lh3/bwa) | `0.7.17-r1188` |
| SAM/BAM processing |  [samtools](https://github.com/samtools/samtools) |  `1.11` | 
| SAM/BAM filtering | [pairtools](https://github.com/open2c/pairtools) | `0.3.0` | 
| Pairs indexing | [pairix](https://github.com/4dn-dcic/pairix) | `0.3.7` | 
| Matrix generation | [cooler](https://github.com/open2c/cooler) | `0.8.10` | 
| Matrix balancing | [hicExplorer](https://github.com/deeptools/HiCExplorer) | `3.6` | 
| Contact map visualization | [HiGlass](http://higlass.io/) | `2.1.11` | 
| Contact map generation | [PretextMap](https://github.com/wtsi-hpag/PretextMap) | `0.1.4` | 
| Contact map visualization | [PretextView](https://github.com/wtsi-hpag/PretextView) | `0.1.5` | 
| Contact map visualization | [PretextSnapshot](https://github.com/wtsi-hpag/PretextSnapshot) | `0.0.3` | 

### Organelle assembly

| Purpose | Program | Version | 
|:-------|:---------|--------:|
| Mitgenome assembly | MitoHiFi |  2 Commit c06ed3e  | 

### Genome quality assessment

| Purpose | Program | Version | 
|:-------|:---------|--------:|
| Basic assembly metrics | [QUAST](https://github.com/ablab/quast) | `5.0.2` | 
| Assembly completeness | [BUSCO](https://busco.ezlab.org/) | `5.0.0` |
| k-mer based assembly evaluation | [Merqury](https://github.com/marbl/merqury) |  `1` | 
| Contamination screening | [BlobToolKit](https://github.com/blobtoolkit/blobtools2) | `2.3.3` |


# Species generated with this pipeline

- *Actinemys marmorata*

[Todd BD, Jenkinson TS, Escalona M, Beraut E, Nguyen O, Sahasrabudhe R, Scott PA, Toffelmier R, Wang IJ, Shaffer HB (2022) ***Reference genome of the northwestern pond turtle, Actinemys marmorata***. Journal of Heredity, 113 (6): 624–631, https://doi.org/10.1093/jhered/esac021](https://academic.oup.com/jhered/article/113/6/624/6602137)

- *Haliotis cracherodii*

[Orland C*, Escalona M* (* Co-first author), Sahasrabudhe R, Marimuthu MPA, Nguyen O, Beraut E, Marshman B, Moore J, Raimondi P, Shapiro B (2022) ***A Draft Reference Genome Assembly of the Critically Endangered Black Abalone, Haliotis cracherodii***. Journal of Heredity, 113 (6): 665–672, https://doi.org/10.1093/jhered/esac024](https://academic.oup.com/jhered/article/113/6/665/6585917)
