# Version(V3.0)

## Overview

## Input
- PacBio HiFi data
- OmniC sequencing data

## Output

The final output corresponds to a diploid assembly, with the primary/alternate approach. 
We are generating 2 pseudo-haplotypes (primary and alternate). The primary assembly is more complete and consists of longer phased blocks. The alternate consists of haplotigs 
(contigs of clones with the same haplotype) in heterozygous regions and is not as complete and more fragmented.
As explained by Heng Li ([here](https://lh3.github.io/2021/04/17/concepts-in-phased-assemblies)), given the characteristics of the latter, it cannot be considered on its own, but as a complement of the primary 
assembly.

The update from previos version relies only on the version of HiFiasm (0.15 to 0.16). See details below.

## Software

Almost all of the links will send you to the corresponding repository. 
The rest of the links correspond to the main documentation web site of the tool.

### Assembly

| Purpose | Program | Version | 
|:-------|:---------|--------:|
| Filtering PacBio HiFi adapters |  [HiFiAdapterFilt](https://github.com/sheinasim/HiFiAdapterFilt) | Commit `64d1c7b` | 
| K-mer counting |  [meryl](https://github.com/marbl/meryl) | `1` | 
| Estimation of genome size and heterozygosity |  [GenomeScope](https://github.com/tbenavi1/genomescope2.0) | `2` |
| *de novo* assembly (contiging) |  [HiFiasm](https://github.com/chhylp123/hifiasm) |   `0.16.1-r375` |
| Long-read, genome-genome alignment |  [minimap2](https://github.com/lh3/minimap2) |  `2.16` | 
| Remove low-coverage, duplicated contigs | [purge_dups](https://github.com/dfguan/purge_dups) | `1.0.1` |
| HiC mapping for SALSA |  [Arima Genomics mapping pipeline](https://github.com/ArimaGenomics/mapping_pipeline) |  Commit `2e74ea4` | 
| HiC Scaffolding | [SALSA](https://github.com/marbl/SALSA) | `2` | 
| Gap closing |  [YAGCloser](https://github.com/merlyescalona/yagcloser) | Commit `20e2769` | 

### Omni-C Contact map generation

| Purpose | Program | Version | 
|:-------|:---------|--------:|
| Short-read alignment  | [bwa](https://github.com/lh3/bwa) | `0.7.17-r1188` |
| SAM/BAM processing |  [samtools](https://github.com/samtools/samtools) |  `1.11` | 
| SAM/BAM filtering | [pairtools](https://github.com/open2c/pairtools) | `0.3.0` | 
| Pairs indexing | [pairix](https://github.com/4dn-dcic/pairix) | `0.3.7` | 
| Matrix generation | [cooler](https://github.com/open2c/cooler) | `0.8.10` | 
| Matrix balancing | [hicExplorer](https://github.com/deeptools/HiCExplorer) | `3.6` | 
| Contact map visualization | [HiGlass](http://higlass.io/) | `2.1.11` | 
| Contact map generation | [PretextMap](https://github.com/wtsi-hpag/PretextMap) | `0.1.4` | 
| Contact map visualization | [PretextView](https://github.com/wtsi-hpag/PretextView) | `0.1.5` | 
| Contact map visualization | [PretextSnapshot](https://github.com/wtsi-hpag/PretextSnapshot) | `0.0.3` | 

### Organelle assembly

| Purpose | Program | Version | 
|:-------|:---------|--------:|
| Mitgenome assembly | [MitoHiFi](https://github.com/marcelauliano/MitoHiFi) |  2 Commit c06ed3e  | 

### Genome quality assessment

| Purpose | Program | Version | 
|:-------|:---------|--------:|
| Basic assembly metrics | [QUAST](https://github.com/ablab/quast) | `5.0.2` | 
| Assembly completeness | [BUSCO](https://busco.ezlab.org/) | `5.0.0` |
| k-mer based assembly evaluation | [Merqury](https://github.com/marbl/merqury) |  `1` | 
| Contamination screening | [BlobToolKit](https://github.com/blobtoolkit/blobtools2) | `2.3.3` |


# Species generated with this pipeline

- *Lynx rufus*

[Lin M, Escalona M, Sahasrabudhe R, Nguyen O, Beraut E, Buchalski MR, Wayne RK (2022) ***A Reference Genome Assembly of the Bobcat, Lynx rufus***. Journal of Heredity, 113 (6): 615–623, https://doi.org/10.1093/jhered/esac031](https://academic.oup.com/jhered/article/113/6/615/6607694)

- *Mytilus californianus*

[Paggeot LX, DeBiasse MB, Escalona M, Fairbairn C, Marimuthu MPA, Nguyen O, Sahasrabudhe R, Dawson MN (2022) Reference genome for the California ribbed mussel, Mytilus californianus, an ecosystem engineer. Journal of Heredity, 113 (6): 681–688, https://doi.org/10.1093/jhered/esac041](https://academic.oup.com/jhered/article/113/6/681/6659931)

- *Embiotoca jacksoni*

[Bernardi G, Toy JA, Escalona M, Marimuthu MPA, Sahasrabudhe R, Nguyen O, Sacco S, Beraut E, Toffelmier E, Miller C, Shaffer HB (2022) Reference genome of the Black Surfperch, Embiotoca jacksoni (Embiotocidae, Perciformes), a California kelp forest fish that lacks a pelagic larval stage. Journal of Heredity, 113 (6): 657–664, https://doi.org/10.1093/jhered/esac034](https://academic.oup.com/jhered/article/113/6/657/6637776)

- *Semicossyphus pulcher*

[Bernardi G, DeBiasse MB, Escalona M, Marimuthu MPA, Nguyen O, Sacco S, Beraut E, Miller C, Toffelmier E, Shaffer HB (2022) ***Reference Genome of the California Sheephead, Semicossyphus pulcher (Labridae, Perciformes), A Keystone Fish Predator in Kelp Forest Ecosystems***. Journal of Heredity, 113 (6): 649-656, https://doi.org/10.1093/jhered/esac032](https://academic.oup.com/jhered/article/113/6/649/6626057)

- Cebidichthys violaceus

[Wright DB, Escalona M, Marimuthu MPA, Sahasrabudhe R, Nguyen O, Sacco S, Beraut E, Toffelmier E, Miller C, Shaffer HB, Bernardi G, German DP. (2023) Reference genome of the Monkeyface Prickleback, Cebidichthys violaceus. Journal of Heredity; 114(1):52-59. https://doi.org/10.1093/jhered/esac054]()

- *Artemisiospiza belli* (Paper in progress) 
- *Melospiza melodia* (Paper in progress)
- *Passerculus sandwichensis* (Paper in progress)
