# Version(V4.0)

## Overview

## Input
- PacBio HiFi data
- OmniC sequencing data

## Output

The final output corresponds to a dual or partially phased diploid assembly (http://lh3.github.io/2021/10/10/introducing-dual-assembly). 

The difference between versions relies on the output sequences from HiFiasm [Version 0.16.1-r375] (Cheng et al. 2021) that are used to generate the final assembly.
We are now using output files `p_ctg.hap1` and `p_ctg.hap2`, and though the genome assemblies are generated as dual or partially phased diploid assemblies, 
we are keeping the naming as primary (for haplotype 1) and alternate (for haplotype 2).


## Software

Almost all of the links will send you to the corresponding repository. 
The rest of the links correspond to the main documentation web site of the tool.

### QC and data preparation

| Purpose | Program | Version | 
|:-------|:---------|--------:|
| Filtering PacBio HiFi adapters |  [HiFiAdapterFilt](https://github.com/sheinasim/HiFiAdapterFilt) | Commit `64d1c7b` | 
| K-mer counting |  [meryl](https://github.com/marbl/meryl) | `1` | 
| Estimation of genome size and heterozygosity |  [GenomeScope](https://github.com/tbenavi1/genomescope2.0) | `2` |

## Assembly 

| Purpose | Program | Version | 
|:-------|:---------|--------:|
| *de novo* assembly (contiging) |  [HiFiasm](https://github.com/chhylp123/hifiasm) |   `0.16.1-r375` |
| Long-read, genome-genome alignment |  [minimap2](https://github.com/lh3/minimap2) |  `2.16` | 
| Remove low-coverage, duplicated contigs | [purge_dups](https://github.com/dfguan/purge_dups) | `1.0.1` |
| HiC mapping for SALSA |  [Arima Genomics mapping pipeline](https://github.com/ArimaGenomics/mapping_pipeline) |  Commit `2e74ea4` | 
| HiC Scaffolding | [SALSA](https://github.com/marbl/SALSA) | `2` | 
| Gap closing |  [YAGCloser](https://github.com/merlyescalona/yagcloser) | Commit `20e2769` | 

### Omni-C Contact map generation

| Purpose | Program | Version | 
|:-------|:---------|--------:|
| Short-read alignment  | [bwa](https://github.com/lh3/bwa) | `0.7.17-r1188` |
| SAM/BAM processing |  [samtools](https://github.com/samtools/samtools) |  `1.11` | 
| SAM/BAM filtering | [pairtools](https://github.com/open2c/pairtools) | `0.3.0` | 
| Pairs indexing | [pairix](https://github.com/4dn-dcic/pairix) | `0.3.7` | 
| Matrix generation | [cooler](https://github.com/open2c/cooler) | `0.8.10` | 
| Matrix balancing | [hicExplorer](https://github.com/deeptools/HiCExplorer) | `3.6` | 
| Contact map visualization | [HiGlass](http://higlass.io/) | `2.1.11` | 
| Contact map generation | [PretextMap](https://github.com/wtsi-hpag/PretextMap) | `0.1.4` | 
| Contact map visualization | [PretextView](https://github.com/wtsi-hpag/PretextView) | `0.1.5` | 
| Contact map visualization | [PretextSnapshot](https://github.com/wtsi-hpag/PretextSnapshot) | `0.0.3` | 

### Organelle assembly

| Purpose | Program | Version | 
|:-------|:---------|--------:|
| Mitgenome assembly | [MitoHiFi](https://github.com/marcelauliano/MitoHiFi) |  2 Commit c06ed3e  | 

### Genome quality assessment

| Purpose | Program | Version | 
|:-------|:---------|--------:|
| Basic assembly metrics | [QUAST](https://github.com/ablab/quast) | `5.0.2` | 
| Assembly completeness | [BUSCO](https://busco.ezlab.org/) | `5.0.0` |
| k-mer based assembly evaluation | [Merqury](https://github.com/marbl/merqury) |  `1` | 
| Contamination screening | [BlobToolKit](https://github.com/blobtoolkit/blobtools2) | `2.3.3` |


# Species generated with this pipeline

- *Pisaster brevispinus*  

[DeBiasse MB, Schiebelhut LM, Escalona M, Beraut E, Fairbairn C, Marimuthu MPA, Nguyen O, Sahasrabudhe R, Dawson MN (2022) A chromosome-level reference genome for the giant pink sea star, Pisaster brevispinus, a species severely impacted by wasting. Journal of Heredity, 113 (6): 689–698, https://doi.org/10.1093/jhered/esac044](https://academic.oup.com/jhered/article/113/6/689/6679504)

- Anthopleura sola

[Cornwell BH, Beraut E, Fairbairn C, Nguyen O, Marimuthu MPA, Escalona M, Toffelmier E(2022) Reference genome assembly of the sunburst anemone, Anthopleura sola. Journal of Heredity, 113 (6): 699–705, https://doi.org/10.1093/jhered/esac050](https://academic.oup.com/jhered/article/113/6/699/6694064)


- Agelaius tricolor

[Ballare KM*, Escalona M* (* Co-first author), Barr K, Seligmann W, Sacco S, Sahasrabudhe R, Nguyen O, Wyckoff C, Smith TB, Shapiro B (2023) A reference genome assembly of the declining tricolored blackbird, Agelaius tricolor. Journal of Heredity, 114(1):44–51, https://doi.org/10.1093/jhered/esac053](https://academic.oup.com/jhered/article/114/1/44/6696806)

- Linanthus parryae

[Anghel I, Jacobs SJ, Escalona M, Marimuthu MPA, Fairbairn CW, Beraut E, Nguyen O, Toffelmier E, Shaffer HB, Zapata F(2022) Reference genome of the color polymorphic desert annual plant Linanthus parryae. Journal of Heredity, 113 (6): 712–721, https://doi.org/10.1093/jhered/esac052](https://academic.oup.com/jhered/article/113/6/712/6701609)

- Haliotis rufescens

[Griffiths JS, Sahasrabudhe RM, Marimuthu MPA, Nguyen OH, Beraut E, Escalona M, Whitehead a (2022) A draft reference genome of the red abalone, Haliotis rufescens, for conservation genomics. Journal of Heredity, 113 (6): 673–680, https://doi.org/10.1093/jhered/esac047](https://academic.oup.com/jhered/article/113/6/673/6747298)

- Clinocottus analis

[Wright DB, Escalona M, Marimuthu MPA, Sahasrabudhe R, Nguyen O, Sacco S, Beraut E, Toffelmier E, Miller C, Shaffer HB, Bernardi G (2023) Reference genome of the Woolly Sculpin, Clinocottus analis. Journal of Heredity, 114 (1):  60–67, https://doi.org/10.1093/jhered/esac055](https://academic.oup.com/jhered/article/114/1/60/6701587)


- *Charina bottae*

[Grismer JL, Escalona M, Miller C, Beraut E, Fairbairn CW, Marimuthu MPA, Nguyen O, Toffelmier E, Wang IJ, Shaffer HB (2022) Reference genome of the Rubber Boa, Charina bottae (Serpentes: Boidae). Journal of Heredity, 113 (6): 641–648, https://doi.org/10.1093/jhered/esac048](https://academic.oup.com/jhered/article/113/6/641/6691355)

- Callipepla californica

[Benham PM, Cicero C, Escalona M, Beraut E, Marimuthu MPA, Nguyen O, Nachman MW, Bowie RCK (2023) A highly contiguous genome assembly for the California quail (Callipepla californica). Journal of Heredity, esad008, https://doi.org/10.1093/jhered/esad008](https://academic.oup.com/jhered/advance-article/doi/10.1093/jhered/esad008/7034149)

- Tetragnatha versicolor

[Adams SA, Graham N, Holmquist A, Sheffer M, Steigerwald EC, Sahasrabudhe R, Nguyen O, Beraut E, Fairbairn C, Sacco S, Seligmann W, Escalona M, Shaffer HB, Toffelmier E, Gillespie RG (2023) Reference genome of the long-jawed orb-weaver, Tetragnatha versicolor (Araneae: Tetragnathidae). Journal of Heredity, esad013. https://doi.org/10.1093/jhered/esad013](https://academic.oup.com/jhered/advance-article/doi/10.1093/jhered/esad013/7115827)

- Cyanocitta stelleri


- Hetaerina americana
- Elgaria multicarinata
- Aquilegia eximia
- Aspidoscelis tigris
- Diadophis punctatus
- Quercus wislizeni
- Sceloporus occidentalis
- Actinemys pallida
- Heteromeles arbutifolia
- Setophaga petechia
- Crotalus oreganus
- Dipodomys merriami
- Melanerpes formicivorus
- Rubus parviflorus
- Nemophila menziesii
