# Version(V4.0)

## Overview

## Input
- PacBio HiFi data
- OmniC sequencing data

## Output

The final output corresponds to a dual or partially phased diploid assembly (http://lh3.github.io/2021/10/10/introducing-dual-assembly). 

The difference between versions is related to a new step that has been added where we attempt to improve the genome assembly
by manually curating both haplotypes using the Rapid Curation toolkit from the Welcome Trust Sanger Institute.

## Software versions

Almost all of the links will send you to the corresponding repository. 
The rest of the links correspond to the main documentation web site of the tool.

### QC and data preparation

| Purpose | Program | Version | 
|:-------|:---------|--------:|
| Filtering PacBio HiFi adapters |  [HiFiAdapterFilt](https://github.com/sheinasim/HiFiAdapterFilt) | Commit `64d1c7b` | 
| K-mer counting |  [meryl](https://github.com/marbl/meryl) | `1` | 
| Estimation of genome size and heterozygosity |  [GenomeScope](https://github.com/tbenavi1/genomescope2.0) | `2` |

## Assembly 

| Purpose | Program | Version | 
|:-------|:---------|--------:|
| *de novo* assembly (contiging) |  [HiFiasm](https://github.com/chhylp123/hifiasm) |   `0.16.1-r375` |
| Long-read, genome-genome alignment |  [minimap2](https://github.com/lh3/minimap2) |  `2.16` | 
| **(Optional)** Remove low-coverage, duplicated contigs | [purge_dups](https://github.com/dfguan/purge_dups) | `1.0.1` |
| HiC mapping for SALSA |  [Arima Genomics mapping pipeline](https://github.com/ArimaGenomics/mapping_pipeline) |  Commit `2e74ea4` | 
| HiC Scaffolding | [SALSA](https://github.com/marbl/SALSA) | `2` | 
| Gap closing |  [YAGCloser](https://github.com/merlyescalona/yagcloser) | Commit `20e2769` | 

### Omni-C Contact map generation

| Purpose | Program | Version | 
|:-------|:---------|--------:|
| Short-read alignment  | [bwa](https://github.com/lh3/bwa) | `0.7.17-r1188` |
| SAM/BAM processing |  [samtools](https://github.com/samtools/samtools) |  `1.11` | 
| SAM/BAM filtering | [pairtools](https://github.com/open2c/pairtools) | `0.3.0` | 
| Pairs indexing | [pairix](https://github.com/4dn-dcic/pairix) | `0.3.7` | 
| Matrix generation | [cooler](https://github.com/open2c/cooler) | `0.8.10` | 
| Matrix balancing | [hicExplorer](https://github.com/deeptools/HiCExplorer) | `3.6` | 
| Contact map visualization | [HiGlass](http://higlass.io/) | `2.1.11` | 
| Contact map generation | [PretextMap](https://github.com/wtsi-hpag/PretextMap) | `0.1.4` | 
| Contact map visualization | [PretextView](https://github.com/wtsi-hpag/PretextView) | `0.1.5` | 
| Contact map visualization | [PretextSnapshot](https://github.com/wtsi-hpag/PretextSnapshot) | `0.0.3` | 

### Organelle assembly

| Purpose | Program | Version | 
|:-------|:---------|--------:|
| Mitgenome assembly | [MitoHiFi](https://github.com/marcelauliano/MitoHiFi) |  2 Commit c06ed3e  | 

### Genome quality assessment

| Purpose | Program | Version | 
|:-------|:---------|--------:|
| Basic assembly metrics | [QUAST](https://github.com/ablab/quast) | `5.0.2` | 
| Assembly completeness | [BUSCO](https://busco.ezlab.org/) | `5.0.0` |
| k-mer based assembly evaluation | [Merqury](https://github.com/marbl/merqury) |  `1` | 
| Contamination screening | [BlobToolKit](https://github.com/blobtoolkit/blobtools2) | `2.3.3` |

### Rapid curation

| Purpose | Program | Version | 
|:-------|:---------|--------:|
| Manual curation | [Rapid curation pipeline (Wellcome Trust Sanger Institute, Genome Reference Informatics Team)](https://gitlab.com/wtsi-grit/rapid-curation)  | Commit 4ddca450 |

# Species generated with this pipeline


- *Phrynosoma blainvillii*

[Richmond JQ, McGuire JA, Escalona M, Marimuthu MPA, Nguyen O, Sacco S, Beraut E, Toffelmier E, Fisher RN, Wang IJ, Shaffer HB (2023) Reference genome of an iconic lizard in western North America, Blainville's horned lizard Phrynosoma blainvillii. Journal of Heredity, esad032. https://doi.org/10.1093/jhered/esad032](https://academic.oup.com/jhered/advance-article/doi/10.1093/jhered/esad032/7169128)
