# Generation and visualization of contact maps

## Overview


Sections below only focus on the generation and visualization of contact maps.

Contact maps, or Hi-C/Omni-C maps, 




## Alignments

## HiGlass contact maps

### Generation

### Visualization

## Pretext contact maps

We generate `pretext` contact maps too because it allow us to identify and solve misassemblies coveniently.


### Generation 

#### Requirements

##### Input

- OmniC to reference alignment (BAM File) sorted by read names (preferably), although the software does not have any requirements for that.


##### Software

- PretextMap
- PretextSnapshot
- samtools
- Conda environment with the PretextSuite installed (download from [here](https://github.com/ccgproject/ccgp_assembly/blob/main/workflows/conda_env/conda.env.pretext.yml))

#### Code

```
conda activate pretext
 samtools view -h -@ $THREADS $WD/aln/assembly.omnic.bam \
    | PretextMap -o $WD/contact_maps/assembly.multimap.pretext \
        --sortby length --sortorder descend &
samtools view -h -@ $THREADS $WD/aln/assembly.pairtools.bam \
    | PretextMap -o $WD/contact_maps/assembly.pairtools.pretext \
        --sortby length --sortorder descend &&
PretextSnapshot -m $WD/contact_maps/assembly.pairtools.pretext \
        -c "Red 2" -r 12000 --gridColour "black" \
        --printSequenceNames \
        -o $WD/contact_maps/ \
        --minTexels 8 \
        --prefix "assembly_" \
        --sequences "=full" &
conda deactivate
```

### Visualization



# Identification of misassemblies

In the following document, you will find information and guidance on how to identify visually potential scenarios of misassemblies.
The document wa generated by the Genome Reference Informatics Team (GRIT) @ the Wellcome Trust Sanger Institute.

ðŸ“„ [Interpreting and using HiC maps for assembly improvement](https://eukaryotic-genome-assembly.github.io/pdf/HiC_Dummies_Guide.pdf)
